tmt/steps/prepare/prepare.py

```diff
@@ -0,0 +1,131 @@
+import dataclasses
+import re
+from typing import List, Optional
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.options
+import tmt.steps
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.provision import Guest
+from tmt.utils import Command, field
+
+
+class FeatureBase(tmt.utils.Common):
+    """ Base class for feature implementations """
+
+    def __init__(
+            self,
+            *,
+            parent: tmt.steps.prepare.PreparePlugin,
+            guest: Guest,
+            logger: tmt.log.Logger) -> None:
+        """ Initialize prepare data """
+        super().__init__(logger=logger, parent=parent, relative_indent=0)
+        self.guest = guest
+
+    def get_system_release(self) -> Optional[str]:
```

Comment:
Could this be a guest fact (obtained in provision)? The `distro:` fact is already gathered there, we just need to make  the output predictable.

Reply:
Hi @lukaszachy and @happz, `guest.facts.distro` is used instead of parsing file `/etc/system-release` on the guest. Function `Feature.get_guest_distro()` is added to reclassify the distro. For example, `Fedora Linux 36` and `Fedora Linux 37` are  reclassified as `Fedora`.